src/components/cc-token-oauth-list/cc-token-oauth-list.js

```diff
@@ -0,0 +1,368 @@
+  iconRemixCalendar_2Fill as iconCreation,
+  iconRemixDeleteBinLine as iconDelete,
+  iconRemixInformationLine as iconExpiration,
+  iconRemixHistoryLine as iconLastUsed,
+} from '../../assets/cc-remix.icons.js';
+import { LostFocusController } from '../../controllers/lost-focus-controller.js';
+import { ResizeController } from '../../controllers/resize-controller.js';
+import { dispatchCustomEvent } from '../../lib/events.js';
+import { isExpirationClose } from '../../lib/tokens.js';
+import { i18n } from '../../translations/translation.js';
+import '../cc-badge/cc-badge.js';
+import '../cc-block/cc-block.js';
+import '../cc-loader/cc-loader.js';
+import '../cc-notice/cc-notice.js';
+
+/**
+ * @typedef {import('./cc-token-oauth-list.types.js').TokenOauthListState} TokenOauthListState
+ * @typedef {import('./cc-token-oauth-list.types.js').OauthTokenState} OauthTokenState
+ * @typedef {import('lit/directives/ref.js').Ref<HTMLDivElement>} RefHTMLDivElement
+ */
+
+/**
+ * A component that displays and manages OAuth tokens.
+ *
+ * This component allows users to view their active OAuth tokens and revoke individual tokens or all tokens at once.
+ * It displays information about each token including consumer name, creation date, last used date, and expiration date.
+ * Tokens are displayed in a list sorted by creation date (newest first).
+ *
+ * @fires {CustomEvent<void>} cc-token-oauth-list:revoke-all-oauth-tokens - Dispatched when a user requests to revoke all tokens
+ * @fires {CustomEvent<string>} cc-token-oauth-list:revoke-oauth-token - Dispatched when a user requests to revoke a specific token, with the token ID as payload
+ */
+export class CcTokenOauthList extends LitElement {
+  static get properties() {
+    return {
+      state: { type: Object },
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @type {TokenOauthListState} The state of the component */
+    this.state = { type: 'loading' };
+
+    /** @type {RefHTMLDivElement} */
+    this._emptyMessageRef = createRef();
+
+    new ResizeController(this, {
+      widthBreakpoints: [995, 730],
+    });
+
+    new LostFocusController(
+      this,
+      '.oauth-token-card__action-revoke, .revoke-all-tokens-button',
+      ({ suggestedElement }) => {
+        if (suggestedElement instanceof HTMLElement) {
+          suggestedElement.focus();
+        } else {
+          this._emptyMessageRef.value.focus();
+        }
+      },
+    );
+  }
+
+  _onRevokeAllOauthTokens() {
+    dispatchCustomEvent(this, 'revoke-all-oauth-tokens');
+  }
+
+  /** @param {string} tokenId */
+  _onRevokeToken(tokenId) {
+    dispatchCustomEvent(this, 'revoke-oauth-token', tokenId);
+  }
+
+  render() {
+    if (this.state.type === 'error') {
+      return html`<cc-notice intent="warning" message="${i18n('cc-token-oauth-list.error')}"></cc-notice>`;
+    }
+
+    const hasOauthTokens =
+      (this.state.type === 'loaded' || this.state.type === 'revoking-all') && this.state.oauthTokens.length > 0;
```

Comment:
nitpick: could also be: `this.state.type === 'revoking-all' || !isEmpty;`

Reply:
No because we also don't want to display the revoking all button when the type is `loading`. If we display the button during the loading phase it creates a weird shift in case you have no tokens (empty list):
- show button during loading,
- hide button once loaded.

It's better to hide the button from the start because it doesn't make sense in loading anyway and hide it only when necessary.